Ejercicio 7.7

1. Escriba un programa que solicite al usuario que ingrese una lista de números enteros. Haz lo siguiente:
b) Imprima el último elemento de la lista.
c) Imprima la lista en orden inverso.
d) Imprima Si, si la lista contiene un 5 y No en caso contrario.
e) Imprima el número de cinco en la lista.
f) Retire el primer y último elemento de la lista, clasifique los elementos restantes e imprima el resultado.
g) Imprima cuántos números enteros en la lista son menores que 5.

In [64]:
num1 = eval(input("Ingrese una lista numeros enteros: "))
l = [num1]
print(l)


[123456]


NOTA: Si ingreso valores sin colocar coma los numeros que me arroja en la lista salen pegados. Ejem: [123456]
      Si ingreso los valores colocando las comas la lista sale asi, [(1, 2, 3, 4, 5, 6)]

In [15]:
# porque se imprime un numero si yo no lo coloco? REVISAR ESTE EJERCICIO
num = eval(input("Ingrese una lista numeros enteros: "))
lista_enteros = []
for i in range(1,10):
    if (i % 1) == 0:
        lista_enteros.append(i)
                       
print(lista_enteros)



[1, 2, 3, 4, 5, 6, 7, 8, 9]


In [76]:
lista = [1, 2, 3, 4, 5]

In [77]:
lista

[1, 2, 3, 4, 5]

In [78]:
# a) Imprima el numero total de elementos de la lista.
len(lista)

5

In [83]:
# b) Imprima el ultimo elemento de la lista.
lista[-1]

5

In [103]:
# c) Imprima la lista en orden inverso.
# NOTA: Tuve que crear la lista para que me arrojara el resultado correcto. 
lista = [1, 2, 3, 4, 5]
lista.reverse()
print(lista)

[5, 4, 3, 2, 1]


In [98]:

# d) Imprima Si, si la lista contiene un 5 y No en caso contrario.
lista = [1, 2, 3, 4, 5, 6]
if (5) in lista:
    print("Si")

else:
    print("No")

Si


In [100]:
# e) Imprima el número de cinco en la lista.
lista = [1, 2, 3, 4, 5, 6]
lista.index(6)


5

In [108]:
# f) Retire el primer y último elemento de la lista, clasifique los elementos restantes e imprima el resultado.
lista = [1, 2, 3, 4, 5, 6]
lista[1:-1]


[2, 3, 4, 5]

In [124]:
# g) Imprima cuántos números enteros en la lista son menores que 5.
lista = [1, 2, 3, 4, 5, 6]
count = 0
for i in lista:
    if i < 5:
        count= count + 1
print(count)
        
    

4


2. Escribe un programa que genere una lista de 20 numeros aleatorios entre 1 y 100.

In [15]:
from random import randint
l = []
for i in range(20):
    l.append(randint(1, 100))

l = l + [randint(1,100)]


In [16]:
# a) Imprima la lista
print(l)

[61, 81, 79, 50, 62, 69, 22, 74, 74, 91, 31, 45, 89, 93, 47, 98, 27, 6, 47, 100, 34]


In [17]:
# b) Imprima el promedio de los elementos de la lista.
suma = sum(l)
num_elem = len(l)
promedio = suma / num_elem
print("El promedio de la lista es:", promedio)

El promedio de la lista es: 60.95238095238095


In [18]:
# c) Imprima los valores mas grandes y mas pequenos de la lista.
l.sort()
print("Valores mas pequenos: ", l[0], l[1])
print("Valores mas grandes: ", l[-1], l[-2])

Valores mas pequenos:  6 22
Valores mas grandes:  100 98


In [19]:
# d) Imprima la segunda entrada mas grande y la segunda mas pequena de la lista.
l.sort()
print(f"Segunda entrada mas grande: {l[-2]}")
print(f"Segunda entrada mas pequena: {l[1]}")

Segunda entrada mas grande: 98
Segunda entrada mas pequena: 22


In [25]:
# e) Imprima cuantos numeros pares hay en la lista.
count = 0
for i in l:
    if i % 2 == 0:
        count += 1 
print(count)       


9


3. Comience con la lista [8,9,10]. Haz lo siguiente:

In [56]:
# a) Establezca la segunda entrada (indice 1) en 17.
lista2 = [8, 9, 10]
lista2[1] = 17
print(lista2)

[8, 17, 10]


In [57]:
# b) Agregue 4, 5 y 6 al final de la lista.
lista2 = [8, 17, 10] + [4, 5, 6]
print(lista2)

[8, 17, 10, 4, 5, 6]


In [58]:
# c) Eliminar la primera entrada de la lista.
del lista2[0]
print(lista2)

[17, 10, 4, 5, 6]


In [59]:
# d) Ordenar la lista.
lista2.sort()
print(lista2)

[4, 5, 6, 10, 17]


In [60]:
# e) Duplicar la lista.
lista2 = lista2*2
print(lista2)

[4, 5, 6, 10, 17, 4, 5, 6, 10, 17]


In [61]:
# f) Insertar 25 en el indice 3.
lista2.insert(3,25)
print(lista2)

[4, 5, 6, 25, 10, 17, 4, 5, 6, 10, 17]


4. Pidale al usuario que ingrese una lista que contenga numeros entre 1 y 12. Luego reemplace todas las entradas en la lista que son mayores de 10 con 10. 

In [4]:
lista_num = []
numeros = eval(input("Introduce una lista de números: "))
for numeros in range(1,13):
    lista_num.append(numeros)
print("Ingresa una lista de numeros entre 1 y 12: ")
print(lista_num)

for i in range(len(lista_num)):
    if lista_num[i] > 10:
        lista_num[i] = 10
        print(lista_num)

Ingresa una lista de numeros entre 1 y 12: 
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 12]
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10]


In [3]:
# Reemplace todas las entradas en la lista que son mayores de 10 con 10.
lista_num

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10]

5. Pida al usuario que ingrese una lista de cadenas. Cree una nueva lista que consta de esas cadenas con sus primeros caracteres eliminados.

In [9]:
num_cadenas = int(input("Introduce el número de cadenas que deseas procesar: "))
listacadenas = []
for i in range(num_cadenas):
    cadena = input(f"Introduce la cadena {i+1}: ")
    listacadenas.append(cadena)

print("La cadena introducida es: ", listacadenas)


La cadena introducida es:  ['CASA', 'CARRO', 'PERRO', 'FAMILIA', 'OFICINA']


In [10]:
listacadenas[2:5]


['PERRO', 'FAMILIA', 'OFICINA']

6. Cree las siguientes listas usando un bucle for.

In [11]:
# a) Una lista que consta de los numeros enteros del 0 al 49.
list_int = []
for number in range(0,50):
    if (number % 1) == 0:
        list_int.append(number)
print(list_int)        

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49]


In [12]:
# b) Una lista que contiene los cuadrados de los números enteros del 1 al 50.list_int = []
list_int1 = []
for number in range(1,51):
    if (number % 1) == 0:
        list_int1.append(number**2)
print(list_int1) 

[1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 400, 441, 484, 529, 576, 625, 676, 729, 784, 841, 900, 961, 1024, 1089, 1156, 1225, 1296, 1369, 1444, 1521, 1600, 1681, 1764, 1849, 1936, 2025, 2116, 2209, 2304, 2401, 2500]


In [ ]:
# c) La lista ['a', 'bb', 'ccc', 'dddd', ...] que termine con 26 copias de la letra z.
